sechub-wrapper-xray/src/test/java/com/mercedesbenz/sechub/wrapper/xray/api/XrayWrapperArtifactoryClientTest.java

```diff
@@ -61,40 +55,37 @@ void getXrayVersion_return_version() throws JsonProcessingException {
     @Test
     void getXrayVersion_throws_xrayWrapperException() {
         /* execute + test */
-        assertThrows(XrayWrapperRuntimeException.class, () -> xrayAPIArtifactoryClient.getXrayVersion());
+        assertThrows(XrayWrapperException.class, () -> xrayAPIArtifactoryClient.getXrayVersion());
     }
 
     @Test
-    void checkArtifactoryUpload_return_true() {
+    void checkArtifactoryUpload_return_true() throws XrayWrapperException {
         /* prepare */
         Mockito.doReturn(response).when(xrayAPIArtifactoryClientSpy).send(any());
 
         /* execute + test */
-        assertTrue(xrayAPIArtifactoryClientSpy.checkArtifactoryUpload());
+        assertTrue(xrayAPIArtifactoryClientSpy.checkArtifactoryUploadSuccess());
     }
 
     @Test
-    void checkArtifactoryUpload_throws_xrayWrapperException() {
+    void checkArtifactoryUpload_throws_xrayWrapperException() throws XrayWrapperException {
         /* prepare */
-        XrayAPIResponse response = XrayAPIResponse.Builder.create(500, headers).build();
+        XrayAPIResponse response = XrayAPIResponse.Builder.builder(500, headers).build();
         Mockito.doReturn(response).when(xrayAPIArtifactoryClientSpy).send(any());
 
         /* execute + test */
-        assertThrows(XrayWrapperRuntimeException.class, () -> xrayAPIArtifactoryClient.checkArtifactoryUpload());
+        assertThrows(XrayWrapperException.class, () -> xrayAPIArtifactoryClient.checkArtifactoryUploadSuccess());
     }
 
     @Test
-    void getScanStatus_return_status_scanned() throws JsonProcessingException {
+    void getScanStatus_return_status_scanned() throws XrayWrapperException {
         /* prepare */
-        ObjectMapper mapper = new ObjectMapper();
         String jsonString = "{\"status\": \"scanned\"}";
-        response = XrayAPIResponse.Builder.create(200, headers).setBody(jsonString).build();
-        JsonNode node = mapper.readTree(jsonString);
+        response = XrayAPIResponse.Builder.builder(200, headers).addResponseBody(jsonString).build();
         Mockito.doReturn(response).when(xrayAPIArtifactoryClientSpy).send(any());
-        Mockito.doReturn(node).when(xrayAPIArtifactoryClientSpy).getBodyAsNode(any());
 
         /* execute */
-        String status = xrayAPIArtifactoryClientSpy.getScanStatus();
+        String status = xrayAPIArtifactoryClientSpy.getScanStatus().getStatus();
```

Comment:
It took a while to understand the test mechanism here.

But ...
You are using a spy object to change the implemenation behavior of the origin part you want to test!

Instead of using
```suggestion
        String status = clientToTest.getScanStatus().getStatus();
```
You are using a spy which overrides the send implementation and return always a dedicated response.

Means: All parts from the send implemenation are overriden and never tested!

IMO the spy approach is not good here. We should discuss a better solution.